test/evergreen/tsan_warnings.supp

```diff
@@ -0,0 +1,9 @@
+# Suppress the listed TSan warnings. Load this file with 
+# export TSAN_OPTIONS="suppressions=path/to/wiredtiger/test/evergreen/tsan_warnings.supp"
+# before running tests with TSan
+
+# FIXME-WT-WT-12512 TSan is raising warnings about mutexes that are non-intuitive. These will be investigated separately.
+mutex:src
+
+# FIXME-WT-WT-12513 The mutex code is a particularly hot path. Review these changes in a separate ticket.
```

Comment:
We'd still be using relaxed atomics here right?

Reply:
My preference would be to fix the race:mtx_rw.c. Also your FIXME's have WT-WT twice.

Reply:
Fixed in [db17785](https://github.com/wiredtiger/wiredtiger/pull/10236/commits/db177850bb66c67cabeed3c889da6c4553dab656). Turns out the mtx_rw changes weren't an issue.

While investigating I realised that we actually see a big perf hit from the `FLD_*` macro changes. I've updated them to only use atomics when building for TSan in [93cce74](https://github.com/wiredtiger/wiredtiger/pull/10236/commits/93cce74a583d8b8c88f9aa6a716a759b641876ce) so there's no perf impacts coming out of WT-12422